sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -353,7 +353,7 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
   test("FORMAT") {
     val f = 'f.string.at(0)
     val d1 = 'd.int.at(1)
-    val s1 = 's.int.at(2)
+    val s1 = 's.string.at(2)
 
     val row1 = create_row("aa%d%s", 12, "cc")
     val row2 = create_row(null, 12, "cc")
```

Comment:
What do we expect, if an Integer value is null? `printf` itself has no problems with null, but for codeGen we have a primitive value like `int` instead of `Integer`. One approach for solving this might be to box all values again and set it to null, if `isNull` return true. Another approach might be to return null if one argument is null